### ISSUE TITLE
Form lists cut off at 200 items; no pagination

### ISSUE THREAD

**Issue by joeflack4** [kobotoolbox/kpi#1983]
## Description
When there are more than 200 items in a form list, it is impossible to view all items without manually searching for items that otherwise aren't appearing using the search bar. There is no pagination.

## Steps to Reproduce
1. Upload more than 200 items to a given list.
2. Observe that the number of rows (specifically of CSS class `asset-row mdl-grid`) in the form list does not exceed 200. (attachment "1" shows this).
3. Observe that there appears to be no way to access additional "pages" of the form list.

## Expected behavior 
When there are more than 200 items in a list, either (a) show all of the items in a single list on a single page, or, preferred, (b) start to paginate the list--typical pagination UI should appear at the bottom of the screen, e.g. a number "1" that is selected, with additional numbers 2 to "n", possibly with additional left/right start/end buttons, with possible ellipses (...) if there are too many pages to display in the bar.

## Actual behavior 
The list gets cut off at 200. It is impossible to view all items without manually searching for items that otherwise aren't appearing using the search bar. There is no pagination.

## Additional details
I am assuming that this problem exists in "projects" and "my library". However I have at this time only confirmed it as being an issue when viewing a "collection"

**_For verification purposes, these images are probably not as helpful as looking at the the html file or reproducing the issue._**

#### Results cut off 1/2
![library results cut off at 200 - 1](https://user-images.githubusercontent.com/13045020/44865537-9ad7df80-ac50-11e8-8eb4-15eeaeb0f641.png)

#### Results cut off 2/2
![library results cut off at 200 - 2](https://user-images.githubusercontent.com/13045020/44865542-9d3a3980-ac50-11e8-9344-f4400cff7281.png)

---

**Attachments**
1. [html file saved from when issue occurred](https://github.com/kobotoolbox/kpi/files/2337015/only200.asset-row.mdl-grid.htm.zip)

**Comment by magicznyleszek** [comment]
Frontend code forces a limit to 200 and currently is not using all endpoint information:

```json
{
  "count": 415,
  "next": "http://192.168.1.15:8000/assets/?limit=200&offset=200&q=asset_type%3Asurvey",
  "previous": null,
  "results": […]
}
```

I think it was assumed that noone will have more than 200 :D

**Comment by jnm** [comment]
:warning: Make sure @jnm adds short-circuits to `filters.py` for any new queries added by the FE: https://github.com/kobotoolbox/kpi/blob/d14e0ba318efddd14a32eb0b6ac4c7e68fc53bef/kpi/filters.py#L135

**Comment by joeflack4** [comment]
> I think it was assumed that noone will have more than 200 ;D 
@magicznyleszek I figured that, lol. We at PMA2020 do =P. Huge users of ODK for 5 years in 13 countries. Soon to add 4 more countries and grant should be renewed probably for the next 10 years, so I think we're going to have a lot!

**Comment by magicznyleszek** [comment]
@joeflack4 `blocked` label means that other issue(s) must be done first, before this one can. In this case, we're waiting for #2009 :)

**Comment by joeflack4** [comment]
Ah, ok!

**Comment by tinok** [comment]
@magicznyleszek For reference, the idea with the form list and library/collection list was actually to load the first 100 assets only, then load the next 100 if the user scrolls down to see them. That was increased to 200 because some users had more than 100 assets. But the real solution here is still outstanding (pagination or load on scroll).

**Comment by magicznyleszek** [comment]
Note for future self - the best way to approach this is to start from #2559 (or wait for merge), and then use `AssetsTable` component as used on `myLibraryRoute.es6`. The overall todo would be:

- [ ] use `assetsTable` in `/forms`
  - [ ] `searches` should become `myForms` and loose all old library related code
  - [ ] `myForms` should handle pagination and other things needed by `assetsTable`
  - [ ] maybe it is easier to write `myForms` from scratch (check where `searches` are being used)
  - [ ] create `myFormsRoute` and use it in `app`
  - [ ] Deployed, Draft and Archived should become a column in AssetsTable?
  - [ ] NOTE: ideally drop all unused pieces of code after all these changes
  - [ ] remember to add contextual empty messages (`{t("Let's get started by creating your first project. Click the New button to create a new form.")}`)
- [ ] use `assetsHash` to not fetch big lists unnecessarily in:
  - [ ] the new store for new `/forms`
  - [ ] `publicCollectionsStore`
  - [ ] `myLibraryStore`
- [ ] make sure `assetNavigator` is still working correctly

**Comment by esario** [comment]
It would be great to have this prioritized, especially given that the public servers now have the [Global P-Codes](https://kf.kobotoolbox.org/#/library/asset/aEjgKH6YNDJ3Cy32t8DVNo) collection of 156 items—56 of them remain hidden.

**Comment by magicznyleszek** [comment]
@esario this was actually working for few years already, but we (I) broke it few months ago. There is a fix waiting to be reviewed: https://github.com/kobotoolbox/kpi/pull/4963. 

As for the issue described in first comment - pagination (infinite scroll) for Projects is already done, so I'm closing this :)
